packages/pipeline-editor/src/PipelineEditor/index.tsx

```diff
@@ -428,13 +428,22 @@ const PipelineEditor = forwardRef(
       []
     );
 
+    const setPropertiesTab = (nodesSelected: boolean) => {
+      if (nodesSelected) {
+        setCurrentTab("properties");
+      } else if (controller.current.getNodes().length > 0) {
+        setCurrentTab("pipeline-properties");
+      } else {
+        setCurrentTab("palette");
+      }
+    };
+
     const handleClickAction = useCallback(
       (e: CanvasClickEvent) => {
         if (e.clickType === "DOUBLE_CLICK" && e.objectType === "node") {
           if (onDoubleClickNode !== undefined) {
             return onDoubleClickNode(e);
           }
-          setCurrentTab("properties");
```

Comment:
why is this removed?

Reply:
I noticed it was redundant while I was working, the first line of the function call below does this.

Reply:
this opens the properties panel when you double click on a node, you wouldn't notice it in elyra because we override the functionality with `onDoubleClickNode` to open the file instead

Reply:
I figured that's what it is for, what I meant was the first line in the properties case in the editActionHandler call in the following line is literally the same line as this. So when the code hits this it will always call setCurrentTab twice. So I removed the redundant line here

Reply:
Oooohhh I see :) I didn’t see that as part of the diff, I thought you were referring to the new logic in selection change handler

nice catch